### PR TITLE
Fix event reliability

### PR DESCRIPTION
**Pull request by mzgoddard**
Reliably integrate events and update the graph rendering when data is missing from missed events.

A common situation is the extension panel is opened after a context is created, we receive a context created event but do not receive events noting creation of nodes, params, and connections. If a node, created after the panel is opened, connects an output to a node, created before the panel opened, data for the pre-opening node will be missing. For example the Audio Destination node will always be one of the created before the panel opened nodes in this situation. This changeset improves integration of events when data is missing from missing earlier events.

- Return early in event integrator if data needed from a prior event is missing
- Create edges in AudioGraphRender only when the needed data from the connecting nodes is present
- Add a fixture, oscillatorGainParam, that creates a context and nodes, and connects a oscillator output to a gain's gain param
- Add unit tests to WebAudioEventIntegrator that uses fixture sequence, oscillatorGainParam, that tests that integration does not create errors while simulating that sequence with all events and simulation of the sequence for each event, filtering out that one event
- Add integration tests that test WebAudioEventIntegrator and AudioGraphRender together and that they do not throw errors on a sequence of events with all the events or with any one event missing. This uses a script built with webpack that `tests/updateGraphRender.ts` opens with puppeteer and checks no errors are thrown.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,4 +6,5 @@ node_modules
 docs
 coverage
 /build
+/simulations/build
```
Comment (hoch) [inline .gitignore:9]: What do "simulations" mean?

File: fixtures/oscillatorGainParam.ts [hunk 2]
```diff
@@ -0,0 +1,385 @@
+import {WebAudioDebuggerEvent} from '../src/chrome/DebuggerWebAudioDomain';
+import {Audion} from '../src/devtools/Types';
+
+export const OSCILLATOR_GAIN_PARAM_EVENTS: Audion.WebAudioEvent[] = [
```
Comment (hoch) [inline fixtures/oscillatorGainParam.ts:4]: What is this for? It looks like a dataset for testing. Can we have an overview comment for this file at the top?

File: simulations/updateGraphRender.html [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<div class="graph" style="height: 100%"></div>
```
Comment (hoch) [inline simulations/updateGraphRender.html:1]: I believe all contents under `simulations` are related to testing. Right?

**Review by hoch — APPROVED** [review]
LGTM with nits

File: simulations/updateGraphRender.html (hunk 3 above)
Comment (mzgoddard) [reply simulations/updateGraphRender.html:1]: Yes. `simulations/updateGraphRender.*` is used by `tests/updateGraphRender.ts`. Simulations like that will need build steps so I was unsure how to immediately file them and put them in a directory.

File: .gitignore (hunk 1 above)
Comment (mzgoddard) [reply .gitignore:9]: Simulations are web pages that run parts of the extension panel from within a normal web page to simulate how those parts behave in the extension panel with some set of devtool protocol input.

**Pull request merged**